### ISSUE TITLE
Falls over if the local Maven repository is large

### ISSUE THREAD

**Issue by argv-minus-one** [diet4j/diet4j#3]
When attempting to launch something with diet4j 0.7, it hangs for a long time (a minute or so), grinding away at the disk, and then crashes with an exception like this:

```
java.io.UncheckedIOException: java.nio.file.FileSystemException: /home/users/alex/.m2/repository/net/sf/jtidy/jtidy/r938: Too many open files
```

A look at your code shows that you are scanning the entire Maven local repository. **_This is an incredibly bad idea.**_ Mine is currently 5.1GB, containing 8,042 folders and 32,783 files. Please don't do this. Examine only the portions needed to locate the requested module and its dependencies.

Since the command line to diet4j includes a groupId and artifactId, you can narrow down your search a great deal. For instance, if the requested module is `com.example:awesome-project`, you need only look at `~/.m2/repository/com/example/awesome-project/maven-metadata-local.xml` to determine the current release version (in the `<metadata>/<versioning>/<release>` element) or the latest version (in the `<metadata>/<versioning>/<versions>/<version>` elements—but beware that they are not sorted, so you must compare them yourself). Then, once you've selected a version (say, 1.0), you need only check that `~/.m2/repository/com/example/awesome-project/1.0/awesome-project-1.0.jar` exists and load it.

I suggest you use [Aether](http://projects.eclipse.org/projects/technology.aether) to do the heavy lifting for you. It can handle resolving transitive dependencies, listing versions, and the like. It only looks in the repositories you explicitly configure, and won't download things without actually being configured with a remote repository to download them from, so it should be safe to use it in diet4j. [The Aether wiki](https://wiki.eclipse.org/Aether) includes a number of useful examples to get you started.

**Comment by jernst** [comment]
You are completely correct for the use case that you are looking at. Patches appreciated :-)
